### PR TITLE
remove torch.cat in look_around()

### PR DESCRIPTION
**Pull request by minh-nguyenhoang**
Resolve new issue in #25

### CODE REVIEW

**Comment by lucidrains** [comment]
@minh-nguyenhoang let's do it!

**Comment by lucidrains** [comment]
@minh-nguyenhoang just so you know, i plan on incorporating [flex attention](https://pytorch.org/blog/flexattention/) into this repo soon

**Comment by minh-nguyenhoang** [comment]
@lucidrains as the current implementation requires many intermediate steps for masking, hoping the `mask_mod` feature can save us some memory and time 😊

**Comment by lucidrains** [comment]
@minh-nguyenhoang oh yea, it will for sure.. flex attention is the future

but your additions are nice too in the interim! thank you!
